src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,286 @@
+import React, { useState } from 'react';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import Courses from '../Courses/Courses';
+import Button from '../../common/Button/Button';
+import Input from '../../common/Input/Input';
+import Textarea from '../../common/Textarea/Textarea';
+
+import { pipeDuration } from '../../helpers/pipeDuration';
+
+/**
+ * Validation
+ */
+let validate = (rule, message) => {
+	if (rule) {
+		return true;
+	}
+	alert(message);
+	return false;
+};
+
+/**
+ * Return all authors
+ *
+ * @param {Object[]} authorsList
+ * @param {Object[]} courseAuthorsList
+ * @returns Object[]
+ */
+let concatAuthors = (authorsList, courseAuthorsList) => {
+	return authorsList.concat(courseAuthorsList);
+};
+
+/**
+ * Author with action button React component
+ */
+function AuthorWithButton(props) {
+	return (
+		<div className='author row my-2'>
+			<div className='name col-6 text-end mt-1'>{props.name}</div>
+			<div className='col-6'>
+				<Button
+					buttonClass={props.buttonClass}
+					buttonText={props.buttonText}
+					onClick={props.onClick}
+				/>
+			</div>
+		</div>
+	);
+}
+
+/**
+ * Create Course React component
+ */
+function CreateCourse(props) {
+	const [coursesList, setCoursesList] = useState(props.coursesList);
+	const [authorsList, setAuthorsList] = useState(props.authorsList);
+	const [courseAuthorsList, setCourseAuthorsList] = useState([]);
+	const [duration, setDuration] = useState('00:00 hours');
+	const [createCoursesHidden, setCreateCoursesHidden] = useState(props.hidden);
+
+	/**
+	 * Create a new author
+	 */
+	let createNewAuthor = () => {
+		// Get Author Name input value
+		let authorName = document.getElementById('createAuthorName').value;
+
+		// Validation
+		let nameValidation = validate(
+			authorName.length >= 2,
+			'Author name length should be at least 2 characters'
+		);
+		if (!nameValidation) return;
+
+		// Copy the Main Authors list
+		let newAuthorsList = authorsList.slice();
+
+		// Add this author to the Main list
+		newAuthorsList.push({
+			id: uuidv4(),
+			name: authorName,
+		});
+
+		// Set new Main Authors list
+		setAuthorsList(newAuthorsList);
+
+		// Clear the input
+		document.getElementById('createAuthorName').value = '';
+	};
+
+	/**
+	 * Add author to the course or remove
+	 *
+	 * @param {string} authorId
+	 * @param {boolean} moveToCourse
+	 */
+	let moveAuthor = (authorId, moveToCourse) => {
+		let aurhorIndex = 0;
+		if (moveToCourse) {
```

Comment:
You're doing same check 3 times in this function, can you join all statements for each case so you would only have 1 check?

Reply:
fixed